### DISCUSSION TITLE
控制台议题：控制台一期后台接口列表

### DISCUSSION THREAD

**Discussion by lightClouds917** [apache/incubator-seata#4113] [Ideas]
### 1. globalSession
- 1.1 PageResult<GlobalSessionVO> queryAll();
- 1.2 PageResult<GlobalSessionVO> queryByStatus(Integer status);
- 1.3 GlobalSessionVO queryByXid(String xid);

### 2. branchSession
- 2.1 PageResult<BranchSessionVO> queryAll();
- 2.2 PageResult<BranchSessionVO> queryByXid(String xid);

### 3. globalLock
- 3.1 PageResult<GlobalLockVO> queryAll();
- 3.2 PageResult<GlobalLockVO> queryByXid(String xid);

**Comment by lvekee** [comment]
### 3. globalLock
* 3.3 PageResult queryAllHangLock();
* 3.4 PageResult queryHangLockByXid(String xid);
* 3.4 Boolean deleteAllHangLock();
* 3.5 Boolean deleteHangLockByKeys(List<String >rowKeys);

**Comment by lvekee** [comment]
### 4.一些想法
* 4.1 可以加一些统计数据的查询，例如，成功了多少次，失败了多少次，全局事务的耗时等，但是这样做的话可能需要增加表结构，需要二阶段在删除记录的同时将这些内容插入到his表中，供控制台查询。

**Reply by funky-eyes** [reply]
> ### 4.一些想法
> * 4.1 可以加一些统计数据的查询，例如，成功了多少次，失败了多少次，全局事务的耗时等，但是这样做的话可能需要增加表结构，需要二阶段在删除记录的同时将这些内容插入到his表中，供控制台查询。

弄个本地内存吧,仅限每个tc当前内存里的记录失败和成功数,想精确的去Grafana和prometheus看

**Reply by lvekee** [reply]
之后考虑直接集成这些在我们的控制台么？

**Reply by funky-eyes** [reply]
> 之后考虑直接集成这些在我们的控制台么？

这些就不必了,未来可以自己记录这些东西,用额外的存储

**Reply by lvekee** [reply]
了解

**Comment by huan415** [comment]
可不可以加个删除死锁的，比如说：我有些本地事务接口接口漏加了GlobalLock+forupdate，造成了脏数据无法回滚，这时候生产环境无法马上改代码，希望能有及时补救的方法（宁愿脏数据，也不要死锁导致用户不能操作）

**Reply by funky-eyes** [reply]
这个会考虑

**Comment by yoster0520** [comment]
弱弱问一句，需要人吗？

**Reply by funky-eyes** [reply]
当然需要,欢迎在此讨论

**Comment by niejian** [comment]
会在哪个版本发布

**Reply by lvekee** [reply]
1.5版本会加入一些简单的接口，后续版本不断迭代

**Comment by huan415** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我最近正在休假中，无法亲自回复你的邮件。我将在假期结束后，尽快给你回复。

**Comment by lightClouds917** [comment]
1.console/globalSession/queryAll   
接口名称：获取全局事物信息列表
方法类型：get
入参：
applicationId    String        应用id
withBranch       boolean    true:带分支事务信息，false:不带分支事务信息
出参：
`{
    "code": "200",
    "message": "success",
    "pageSize": 0, //每页数据量
    "pageNum": 0, //页码
    "total": 2,//数据总量
    "pages": 0,//总页数
    "data": [
        {
            "xid": null, //全局事务id
            "transactionId": null, //事务id
            "status": null, //全局事务状态
            "applicationId": null,//应用id
            "transactionServiceGroup": null,//事务分组
            "transactionName": "tst1",//事务名称
            "timeout": null,//事务超时时间
            "beginTime": null,//事务开始时间
            "applicationData": null,//应用数据
            "gmtCreate": null,//记录创建时间
            "gmtModified": null,//记录修改时间
            "branchSessionVOs": [//分支事物列表
                {
                    "xid": null,//全局事务id
                    "transactionId": null,//事务id
                    "branchId": null,//分支事务id
                    "resourceGroupId": null,//事务分组id
                    "resourceId": null,//资源id
                    "branchType": null,//分支类型
                    "status": null,//分支事务状态
                    "clientId": null,//客户端id
                    "applicationData": null,//应用数据
                    "gmtCreate": null,//记录创建时间
                    "gmtModified": null//记录修改时间
                },
                {
                    "xid": null,
                    "transactionId": null,
                    "branchId": null,
                    "resourceGroupId": null,
                    "resourceId": null,
                    "branchType": null,
                    "status": 2,
                    "clientId": null,
                    "applicationData": "ferw",
                    "gmtCreate": null,
                    "gmtModified": null
                }
            ]
        },
        {
            "xid": null,
            "transactionId": null,
            "status": null,
            "applicationId": null,
            "transactionServiceGroup": null,
            "transactionName": "tst2",
            "timeout": null,
            "beginTime": null,
            "applicationData": null,
            "gmtCreate": null,
            "gmtModified": null,
            "branchSessionVOs": null
        }
    ],
    "currPage": 0,//当前页码
    "success": true
}`
